File: java-prov/src/main/java/cz/muni/fi/bthesis/IdentifierWithoutPrefix.java
package cz.muni.fi.bthesis;

import org.openprovenance.prov.model.Entity;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.vanilla.Document;
import org.openprovenance.prov.vanilla.ProvFactory;

/**
 * @author Matus Formanek
 */
public class IdentifierWithoutPrefix extends TestCase{
    public IdentifierWithoutPrefix(){
        setFilename("identifier_without_prefix");
    }
    @Override
    public Document createDocument() {
        ProvFactory factory = new ProvFactory();
        var document = new org.openprovenance.prov.vanilla.Document();
        var ns = new Namespace();
        ns.addKnownNamespaces();
        ns.registerDefault("https://default.org/");
        var e = ns.qualifiedName(null, "e", factory);
        Entity entity = factory.newEntity(e);
        document.getStatementOrBundle().add(entity);
        document.setNamespace(ns);
        return document;
    }
}

File: python-prov/src/identifier_without_prefix.py
from prov.model import ProvDocument

from test_case import TestCase


class IdentifierWithoutPrefix(TestCase):

    def __init__(self):
        self.filename = "identifier_without_prefix"

    def create_document(self):
        document = ProvDocument()
        document.set_default_namespace("https://default.org/")
        document.entity("e")
        return document
